### PR TITLE
feat: append whitespace after inserting emoji

### PR DESCRIPTION
**Pull request by aswin-s**
### Details

This PR fixes below regressions in https://github.com/Expensify/App/pull/30412
1. Additional whitespace was added to a pasted emoji even if it already had a leading or trailing white space. 
2. Emoji's with skin tones were getting corrupted after appending white space. 

Discussion here: https://github.com/Expensify/App/issues/23658#issuecomment-1797273930

### Fixed Issues

$ https://github.com/Expensify/App/issues/23658
PROPOSAL: 

### Tests

1. Open any chat
2. Enter an emoji via native keyboard, emoji picker or short code (eg- `:heart:`)
3. Verify that the emoji gets appended by a white space.
5. Verify that emoji can be removed by tapping backspace.
6. Verify cursor is positioned properly after inserting emoji.
7.  Verify that emoji can be inserted at the midst of text
8. Verify that an emoji can be copy pasted
9. Verify that complex emoji like '👨‍👩‍👧‍👦' can be added and removed.
10. Verify that while pasting text, white space is added only when emoji is pasted. Whitespace shouldn't be added if the pasted text has any other characters other than emoji, including whitespace.
11. Verify that emojis with different skin tones could be inserted

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Open any chat
2. Enter an emoji via native keyboard, emoji picker or short code (eg- `:heart:`)
3. Verify that the emoji gets appended by a white space.
4. Verify that emoji can be removed by tapping backspace.
5. Verify cursor is positioned properly after inserting emoji.
6.  Verify that emoji can be inserted at the midst of text
7. Verify that an emoji can be copy pasted
8. Verify that complex emoji like '👨‍👩‍👧‍👦' can be added and removed.
9. Verify that while pasting text, white space is added only when emoji is pasted. Whitespace shouldn't be added if the pasted text has any other characters other than emoji, including whitespace.
10. Verify that emojis with different skin tones could be inserted

### QA Steps

1. Open any chat
2. Enter an emoji via native keyboard, emoji picker or short code (eg- `:heart:`)
3. Verify that the emoji gets appended by a white space.
4. Verify that emoji can be removed by tapping backspace.
5. Verify cursor is positioned properly after inserting emoji.
6.  Verify that emoji can be inserted at the midst of text
7. Verify that an emoji can be copy pasted
8. Verify that complex emoji like '👨‍👩‍👧‍👦' can be added and removed.
9. Verify that while pasting text, white space is added only when emoji is pasted. Whitespace shouldn't be added if the pasted text has any other characters other than emoji, including whitespace.
10. Verify that emojis with different skin tones could be inserted

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/6880914/52e5b454-9171-4029-b290-65eedc8d623a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/6880914/3cade01f-843b-49fe-9ca4-9e162a1de602

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/6880914/aaf2723c-3d63-4b91-b41c-86a8bfffc37f

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/6880914/ff1cd89f-ac7c-470f-ba4e-4db032527e3d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/6880914/45bd0ad0-c94b-4f1e-92db-a30d735b07fd

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/6880914/94f6abb0-d815-4d84-b9cf-ddbdb3eb6641
</details>

### CODE REVIEW

**Comment by mollfpr** [comment]
@aswin-s Are this PR fixing this https://github.com/Expensify/App/issues/30983 deploy blocker?

**Comment by aswin-s** [comment]
> @aswin-s Are this PR fixing this #30983 deploy blocker?

Nope, it handles extra white space appended when the pasted content already contains a trailing whitespace.

**Comment by mollfpr** [comment]
@aswin-s Could you pull the latest main? Thank you!

**Comment by aswin-s** [comment]
@mollfpr Done!

**Comment by mollfpr** [comment]
>1. Additional whitespace was added to a pasted emoji even if it already had a leading or trailing white space.
>2. Emoji's with skin tones were getting corrupted after appending white space.

Also, I need help figuring out the changes for the above fix. Could you explain it more? Thanks!

**Comment by aswin-s** [comment]
@mollfpr Here is the explanation for the changes

> 1. Additional whitespace was added to a pasted emoji even if it already had a leading or trailing white space.

The earlier logic didn't consider if the diff from previous text had leading or trailing white space, there by appending an additional white space after emoji. This was fixed by checking if the diff indeed has leading or trailing white space and skipping the addition of white space if there is any.

Relevant Commit: https://github.com/Expensify/App/pull/30955/commits/639136ad63796d54e7c18af31b6cd769fcd4f9e4

> Emoji's with skin tones were getting corrupted after appending white space.

This issue occurred because of appending whitespace immediately after a detected emoji character. However for emojis with skin tone immediately after the actual emoji there were additional unicode characters that denoted what the skin tone is. By simply adding a white space after the original emoji and then followed by the skin tone unicode characters the emoji itself got corrupted. In the fix I made sure that the entire emoji length is taken into consideration before appending white space.
 
Relevant Commit: https://github.com/Expensify/App/pull/30955/commits/6062753ff9e20bae3f6f1ca1623c55c4f63ff1c3

**Comment by aswin-s** [comment]
@mollfpr Waiting for your feedback

**Comment by mollfpr** [comment]
@aswin-s Most cases seem to be working fine, but I have one issue with pasting only an emoji next to another, which doesn't add a whitespace after it. It works fine by pasting an emoji next to a text.


https://github.com/Expensify/App/assets/25520267/d28ca1a4-d806-4e15-a40c-cb13ca8f943e

Also, could you resolve the conflict? Thanks!

**Comment by aswin-s** [comment]
@mollfpr Fixed and retested all scenarios.

https://github.com/Expensify/App/assets/6880914/5d7e5c4b-53ee-43b7-ba50-c19bc5b885ca

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by mollfpr — APPROVED** [review]
LGTM 🚀

File: src/libs/ComposerUtils/index.ts [hunk 1]
```diff
@@ -23,4 +34,26 @@ function canSkipTriggerHotkeys(isSmallScreenWidth: boolean, isKeyboardShown: boo
...
 
-export {getNumberOfLines, updateNumberOfLines, insertText, canSkipTriggerHotkeys};
+/**
+ * Finds the length of common suffix between two texts
+ * @param str1 - first string to compare
+ * @param str2 - next string to compare
+ * @param cursorPosition - position of cursor
+ * @returns number - Length of the common suffix
+ */
+function findCommonSuffixLength(str1: string, str2: string, cursorPosition: number) {
```
Comment (flodnv) [inline src/libs/ComposerUtils/index.ts:44]: This feels like it belongs in expensify/common str.js, no?

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js [hunk 2]
```diff
@@ -312,14 +361,8 @@ function ComposerWithSuggestions({
      * @param {Boolean} shouldAddTrailSpace
```
Comment (flodnv) [inline src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js:361]: Did we get rid of this?

File: src/libs/ComposerUtils/index.ts [hunk 3]
```diff
@@ -14,6 +14,17 @@ function insertText(text: string, selection: Selection, textToInsert: string): s
     return text.slice(0, selection.start) + textToInsert + text.slice(selection.end, text.length);
 }
 
+/**
+ * Insert a white space at given index of text
+ * @param text - text that needs whitespace to be appended to
+ * @param index  - index at which whitespace should be inserted
+ * @returns
```
[the remark this answers is not in the document]
Comment (aswin-s) [reply src/libs/ComposerUtils/index.ts]: Removed

File: src/libs/ComposerUtils/index.ts [hunk 4]
```diff
@@ -23,4 +34,26 @@ function canSkipTriggerHotkeys(isSmallScreenWidth: boolean, isKeyboardShown: boo
     return (isSmallScreenWidth && DeviceCapabilities.canUseTouchScreen()) || isKeyboardShown;
 }
 
-export {getNumberOfLines, updateNumberOfLines, insertText, canSkipTriggerHotkeys};
+/**
+ * Finds the length of common suffix between two texts
+ * @param str1 - first string to compare
+ * @param str2 - next string to compare
+ * @param cursorPosition - position of cursor
+ * @returns number - Length of the common suffix
```
[the remark this answers is not in the document]
Comment (aswin-s) [reply src/libs/ComposerUtils/index.ts]: Removed

File: src/libs/ComposerUtils/index.ts (hunk 1 above)
Comment (aswin-s) [reply src/libs/ComposerUtils/index.ts:38]: At the moment it is used only with composer and requires cursor position to exactly predict which character got appended when duplicate characters are present.

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js (hunk 2 above)
Comment (aswin-s) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js:361]: @flodnv Didn't quite get the question. Do you mean did we get rid of `@param` from jsdoc? I think it has become redundant with Typescript migration.

File: src/libs/ComposerUtils/index.ts [hunk 5]
```diff
@@ -14,6 +14,15 @@ function insertText(text: string, selection: Selection, textToInsert: string): s
     return text.slice(0, selection.start) + textToInsert + text.slice(selection.end, text.length);
 }
 
+/**
+ * Insert a white space at given index of text
+ * @param text - text that needs whitespace to be appended to
+ */
+
```
Comment (flodnv) [inline src/libs/ComposerUtils/index.ts:21]: unnecessary empty line

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js (hunk 2 above)
Comment (flodnv) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js:361]: Sorry, I am asking: it seems that we got rid of the `shouldAddTrailSpace` variable in code, and if I'm correct it seems like we should remove this `@param` line?

**Review by flodnv — COMMENTED** [review]
Looks good, one more thought: is this code performing well on an input of 100k chars?

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js (hunk 2 above)
Comment (aswin-s) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js]: Removed the `@param` line

**Comment by aswin-s** [comment]
> Looks good, one more thought: is this code performing well on an input of 100k chars?

Yep, tested with 100K characters and performing well.

**Review by flodnv — APPROVED** [review]

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js [hunk 6]
```diff
@@ -198,7 +240,13 @@ function ComposerWithSuggestions({
...
         (commentValue, shouldDebounceSaveComment) => {
             raiseIsScrollLikelyLayoutTriggered();
-            const {text: newComment, emojis, cursorPosition} = EmojiUtils.replaceAndExtractEmojis(commentValue, preferredSkinTone, preferredLocale);
+            const {startIndex, endIndex, diff} = findNewlyAddedChars(lastTextRef.current, commentValue);
+            const isEmojiInserted = diff.length && endIndex > startIndex && diff.trim() === diff && EmojiUtils.containsOnlyEmojis(diff);
+            const {
+                text: newComment,
+                emojis,
+                cursorPosition,
+            } = EmojiUtils.replaceAndExtractEmojis(isEmojiInserted ? ComposerUtils.insertWhiteSpaceAtIndex(commentValue, endIndex) : commentValue, preferredSkinTone, preferredLocale);
```
Comment (eh2077) [inline src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js:249]: This line caused a regression [iOS - Chat - The cursor moves one space backward when inserting text after an emoji](https://github.com/Expensify/App/issues/42664#top)

**Pull request merged**